zerver/lib/bugdown/__init__.py

```diff
@@ -1011,10 +1011,17 @@ def run(self, root: Element) -> None:
         if len(found_urls) == 0 or len(found_urls) > self.INLINE_PREVIEW_LIMIT_PER_MESSAGE:
```

Comment:
Oh, actually, one question -- should we be doing this number of URLs check after deduplication?

Reply:
Yeah, I think that makes sense! I'll update the PR and let you know.